src/Symfony/Component/Validator/Tests/Constraints/NotCompromisedPasswordValidatorTest.php

```diff
@@ -225,44 +227,27 @@ public static function provideErrorSkippingConstraints(): iterable
 
     private function createHttpClientStub(?string $returnValue = null): HttpClientInterface
     {
-        $httpClientStub = $this->createMock(HttpClientInterface::class);
-        $httpClientStub->method('request')->willReturnCallback(
-            function (string $method, string $url) use ($returnValue): ResponseInterface {
-                if (self::PASSWORD_TRIGGERING_AN_ERROR_RANGE_URL === $url) {
-                    throw new class('Problem contacting the Have I been Pwned API.') extends \Exception implements ServerExceptionInterface {
-                        public function getResponse(): ResponseInterface
-                        {
-                            throw new \RuntimeException('Not implemented');
-                        }
-                    };
-                }
-
-                $responseStub = $this->createMock(ResponseInterface::class);
-                $responseStub
-                    ->method('getContent')
-                    ->willReturn($returnValue ?? implode("\r\n", self::RETURN));
-
-                return $responseStub;
+        return new MockHttpClient(function ($method, $url) {
+            if (self::PASSWORD_TRIGGERING_AN_ERROR_RANGE_URL === $url) {
+                throw new class('Problem contacting the Have I been Pwned API.') extends \Exception implements ServerExceptionInterface {
```

Comment:
Why not throwing directly the class of the http-client component (the usage of MockHttpClient introduced a hard dependency anyway)